src/vis_cpu/conversions.py

```diff
@@ -258,7 +267,57 @@ def equatorial_to_eci_coords(ra, dec, obstime, location, unit="rad", frame="icrs
     return pra, pdec
 
 
-def uvbeam_to_lm(uvbeam, freqs, n_pix_lm=63, polarized=False, **kwargs):
+def prepare_beam(
+    uvbeam: UVBeam, polarized: bool = False, use_feed: Literal["x", "y"] = "x"
+) -> UVBeam:
+    """Prepare an imput beam for either interpolation or simulation.
+
+    The point of this function is to take an arbitrary UVBeam (or AnalyticBeam) and
+    do the necessary checks and conversions to convert it to a format that can be
+    interpolated to an (l,m) grid, or passed to vis_cpu. The output beam type is
+    dependent on the input parameters ``polarized`` and ``use_feed``.
+    """
+    use_feed = use_feed.lower()
+    if use_feed not in "xy":
+        raise ValueError("use_feed must be either 'x' or 'y'")
+    use_pol = use_feed * 2
+
+    # Interpolate beam onto cube
+    if polarized:
+        if uvbeam.beam_type != "efield":
+            raise ValueError("Beam type must be efield")
+        uvbeam_ = uvbeam
+    elif uvbeam.beam_type == "efield":
+        uvbeam_ = uvbeam.copy() if isinstance(uvbeam, UVBeam) else deepcopy(uvbeam)
+        # Analytic beams have no concept of feeds, so assume they have a "single" feed
+        if getattr(uvbeam_, "Nfeeds", 1) > 1:
+            uvbeam_.select(feeds=[use_feed])
+
+        uvbeam_.efield_to_power()
+
+    elif getattr(uvbeam, "Npols", 1) > 1:
+        pol = uvutils.polstr2num(use_pol)
+
+        if pol not in uvbeam.polarization_array:
+            raise ValueError(
+                f"You want to use {use_feed} feed, but it does not exist in the UVBeam"
+            )
```

Comment:
I don't think we need to include this check? It looks like this is something that's already covered in the `select` method.

Reply:
I'd still prefer to keep this here to make it more obvious for the user.